ding/entry/application_entry_drex_collect_data.py

```diff
@@ -0,0 +1,253 @@
+        Collect episodic demonstration data by the trained policy for trex specifically.
+    Arguments:
+        - cfg (:obj:`easydict`): Config in dict type.
+        - seed (:obj:`int`): Random seed.
+        - collect_count (:obj:`int`): The count of collected data.
+        - rank (:obj:`int`) the episode ranking.
+        - save_cfg_path(:obj:'str') where to save the collector config
+        - env_setting (:obj:`Optional[List[Any]]`): A list with 3 elements: \
+            ``BaseEnv`` subclass, collector env config, and evaluator env config.
+        - model (:obj:`Optional[torch.nn.Module]`): Instance of torch.nn.Module.
+        - state_dict (:obj:`Optional[dict]`): The state_dict of policy or model.
+        - state_dict_path (:obj:'str') the abs path of the state dict
+    """
+    cfg.env.collector_env_num = 1
+    if not os.path.exists(save_cfg_path):
+        os.mkdir(save_cfg_path)
+
+    # Create components: env, policy, collector
+    if env_setting is None:
+        env_fn, collector_env_cfg, _ = get_vec_env_setting(cfg.env)
+    else:
+        env_fn, collector_env_cfg, _ = env_setting
+    collector_env = create_env_manager(cfg.env.manager, [partial(env_fn, cfg=c) for c in collector_env_cfg])
+    collector_env.seed(seed)
+
+    set_pkg_seed(seed, use_cuda=cfg.policy.cuda)
+    policy = create_policy(cfg.policy, model=model, enable_field=['collect', 'eval'])
+    collect_demo_policy = policy.collect_mode
+    if state_dict is None:
+        assert state_dict_path is not None
+        state_dict = torch.load(state_dict_path, map_location='cpu')
+    policy.collect_mode.load_state_dict(state_dict)
+    collector = EpisodeSerialCollector(cfg.policy.collect.collector, collector_env, collect_demo_policy)
+
+    policy_kwargs = {'eps': noise}
+
+    # Let's collect some sub-optimal demonstrations
+    exp_data = collector.collect(n_episode=collect_count, policy_kwargs=policy_kwargs)
+
+    if cfg.policy.cuda:
+        exp_data = to_device(exp_data, 'cpu')
+    # Save data transitions.
+    print('Collect {}th episodic demo data successfully'.format(rank))
+    return exp_data
+
+
+def drex_get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--cfg', type=str, default='abs path for a config')
+    parser.add_argument('--seed', type=int, default=0)
+    parser.add_argument('--device', type=str, default='cuda' if torch.cuda.is_available() else 'cpu')
+    args = parser.parse_known_args()[0]
+    return args
+
+
+def eval_bc(validation_set, policy, use_cuda):
+    tot = 0
+    tot_acc = 0
+    device = 'cuda' if use_cuda else 'cpu'
+    for _, data in enumerate(validation_set):
+        x, y = {'obs': data['obs'].to(device).squeeze(0)}, data['action'].squeeze(-1)
+        y_pred = policy.forward(x, eps=-1)['obs']['action']
+        tot += y_pred.shape[0]
+        tot_acc += (y_pred == y).sum().item()
+    acc = tot_acc / tot
+    return acc
+
+
+def train_bc(cfg, pre_expert_data=None, max_iterations=6000):
```

Comment:
merge `train_bc` and `evaluate_bc` functions into `serial_pipeline_il` (or you can rename it to `serial_pipeline_bc`)